### PR TITLE
use templates to create function project

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/resources/azurefunction/templates/project/build.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+sbt.version=1.2.8
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/resources/azurefunction/templates/project/build.properties]: Is this file used?

File: PluginsAndFeatures/azure-toolkit-for-intellij/resources/azurefunction/templates/project/build_gradle.template [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    pricingTier = 'Consumption' // refers https://github.com/microsoft/azure-maven-plugins/tree/develop/azure-functions-maven-plugin#supported-pricing-tiers for all valid values
+    region = 'westus'
+    runtime {
+      os = 'windows'
+    }
+    // appSettings {
+    //    key = 'value'
+    // }
+    // localDebug = "transport=dt_socket,server=y,suspend=n,address=5005"
+}
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/resources/azurefunction/templates/project/build_gradle.template]: Empty line at the end

File: PluginsAndFeatures/azure-toolkit-for-intellij/resources/azurefunction/templates/project/pom_xml.template [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+    <version>$(version)</version>
+    <packaging>jar</packaging>
+
+    <name>Azure Java Functions</name>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
+        <azure.functions.maven.plugin.version>1.4.0</azure.functions.maven.plugin.version>
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/resources/azurefunction/templates/project/pom_xml.template]: latest is 1.4.1, besides, do we need a pipeline to update the dependency versions automatically?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/util/AzureFunctionsUtils.java [hunk 4]
```diff
@@ -83,43 +80,51 @@ public static String normalizeClassName(String className) {
...
-            return folder;
+    public static File createFunctionProjectToTempFolder(final String groupId, final String artifactId,
+                                                         final String version, final String tool)
+        throws IOException {
+        final File folder = Files.createTempDir();
+        final Map<String, String> map = new HashMap<>();
+        map.put("groupId", groupId);
+        map.put("artifactId", artifactId);
+        map.put("version", version);
+        map.put("randomNumber", Long.toString(new Date().getTime()));
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/util/AzureFunctionsUtils.java]: randomNumber -> timestamp?

File: PluginsAndFeatures/azure-toolkit-for-intellij/resources/azurefunction/templates/project/pom_xml.template (hunk 3 above)
Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-intellij/resources/azurefunction/templates/project/pom_xml.template]: will be handled later

File: PluginsAndFeatures/azure-toolkit-for-intellij/resources/azurefunction/templates/project/build.properties (hunk 1 above)
Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-intellij/resources/azurefunction/templates/project/build.properties]: no, will remove it

**Pull request merged**